streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -1225,8 +1225,16 @@
      * @return a {@link KStream} containing all records of the given streams
      */
     public <K, V> KStream<K, V> merge(final KStream<K, V>... streams) {
+        Objects.requireNonNull(streams, "streams can't be null");
+        if (streams.length <= 1) {
```

Comment:
just to note, in case of length == 1 your logic will work fine (and old implementation handles this case)

I am thinking would it simplify user's code if 1 stream could be accepter?

Reply:
Merging 1 stream only is not normal as it will just be a no-op, I think making it more strict in the new code makes more sense.

BTW we have been deprecating KStreamBuilder and we are moing `merge()` to `KStream` which will take one stream at a time.